### PR TITLE
test-snyk

### PR DESCRIPTION
**Pull request by mikicho**

### CODE REVIEW

**Review by goldbergyoni — DISMISSED** [review]
@mikicho Looks good to me

@lirantal Other optimizations that we should be aware of? 

Both - Can we include it also for the users (the generated code) or do they need to generate personal token?

**Comment by lirantal** [comment]
This one runs the SCA check, i.e: dependencies.
You could add another step with a custom `command` of `code test` which will also test for security issue in your own code, i.e: SAST.

**Comment by mikicho** [comment]
@lirantal added SAST, I'm just not sure I added it correctly:
https://github.com/practicajs/practica/pull/151/files#diff-6102d4f2cddb56c446459bde9bf8e8044b87ce3f98a24c2bba99debfd62461dcR21

In the build, I don't see an indication for SCA, only for SAST (`Test type:         Static code analysis`):
image

Link to build: https://github.com/practicajs/practica/runs/7476884507?check_suite_focus=true

**Comment by lirantal** [comment]
@mikicho correct, because the `command` directive overrides the default `test` with `code test` which means you replace the default SCA with the SAST one.

So, to have both, duplicate that job so that each runs a different scan (parallelism), or you can add another step.

**Comment by lirantal** [comment]
Looks like it is working now, and also that you have a couple of findings: https://github.com/practicajs/practica/runs/7480982678?check_suite_focus=true :-)

I would go ahead and add a .snyk file to ignore those vulns as they seem to be impacting your dev dependencies, right?

Another thing to flag is that you may want to customize the SCA part with this command argument `--all-projects` (add: `command: test --all-projects`) in order to scan across monorepos and other package manifest locations.

**Comment by mikicho** [comment]
@lirantal added `--all-project` option.
I'm trying to produce a sarif file for both SAST and SCA but I get this error, is this a bug?
image

https://github.com/practicajs/practica/runs/7483982273?check_suite_focus=true

**Comment by goldbergyoni** [comment]
@mikicho Maybe merge the CVE scan first then we will figure out the static code analysis?

**Comment by lirantal** [comment]
@mikicho this one is incorrect:
https://github.com/practicajs/practica/actions/runs/2725251965/workflow#L27

It should be `code test`

**Comment by mikicho** [comment]
@lirantal why I can't find `code test` command in [the docs](https://docs.snyk.io/snyk-cli/cli-reference#snyk-code)?

**Comment by mikicho** [comment]
@goldbergyoni I fixed the snyk part :) 
@lirantal I disable this one because I already added this check in the `github actions`. p.s: Now I'm not sure what the advantage of the github action if it's already check for vulns and do the static check for me automatically (?)
image

**Comment by lirantal** [comment]
> @lirantal why I can't find `code test` command in [the docs](https://docs.snyk.io/snyk-cli/cli-reference#snyk-code)?

It's there, if you click into `snyk code` in that list, you get here: https://docs.snyk.io/snyk-cli/commands/code which mentions `snyk code test` as a subcommand.

> @goldbergyoni I fixed the Snyk part :) why does the coverage fail? :( @lirantal I disabled this one because I already added this check in the `github actions`. p.s: Now I'm not sure what the advantage of the Snyk (GitHub) action is if it already checks for vulns and do the static check for me automatically (?) image

The value of the native GitHub integration into Snyk is quite high:
1. You get automatically monitoring your project. GitHub Actions only run when they're executed. Unless you explicitly set some type of cronjob for them, they could theoretically be stalled for months until a new commit or other trigger causes them to execute. You could miss many vulnerabilities during this time frame.
2. The native GitHub integration allows the Snyk bot to open automatic PRs to fix and suggest package upgrades and Dockerfile fixes
3. You can enable status checks as part of the PR and how it behaves (allow it to block the PR if you want to) which is unlike a GitHub Action execution that could be an all-or-nothing, depending on how you built it.

File: docs/package.json [hunk 1]
```diff
@@ -14,9 +14,9 @@
     "write-heading-ids": "docusaurus write-heading-ids"
   },
   "dependencies": {
-    "@docusaurus/core": "2.0.0-beta.18",
+    "@docusaurus/core": "^2.1.0",
```
Comment (goldbergyoni) [inline docs/package.json:17]: @mikicho  This is a major change, could you check if /docs 'npm run build' and 'npm start' works?

We might break our website

**Comment by goldbergyoni** [comment]
@mikicho I could not understand why codecov is angry. Since we know that there is no code logic here or coverage drop, let's ignore this status fail. Do you have enough rights?

File: docs/package.json (hunk 1 above)
Comment (mikicho) [reply docs/package.json:17]: check

**Comment by mikicho** [comment]
> @mikicho I could not understand why codecov is angry. Since we know that there is no code logic here or coverage drop, let's ignore this status fail. Do you have enough rights?

@goldbergyoni I'm afraid it will break the ci because it will fail later PRs, am I wrong?

**Comment by mikicho** [comment]
Ok lets merge this, and decide later about the snyk github check, (the integration is working, I just disabled the PR check)

**Pull request merged**